pkg/planner/core/plan_cache.go

```diff
@@ -116,10 +116,11 @@ func planCachePreprocess(ctx context.Context, sctx sessionctx.Context, isNonPrep
 				return plannererrors.ErrSchemaChanged.GenWithStack("Schema change caused error: %s", err.Error())
 			}
 			delete(stmt.RelateVersion, stmt.tbls[i].Meta().ID)
+			tbl = tblByName
 			stmt.tbls[i] = tblByName
 			stmt.RelateVersion[tblByName.Meta().ID] = tblByName.Meta().Revision
 		}
-		newTbl, err := tryLockMDLAndUpdateSchemaIfNecessary(ctx, sctx.GetPlanCtx(), stmt.dbName[i], stmt.tbls[i], is)
+		newTbl, err := tryLockMDLAndUpdateSchemaIfNecessary(ctx, sctx.GetPlanCtx(), stmt.dbName[i], tbl, is)
```

Comment:
Could we add some comments for this change(using `tbl`)?

Reply:
tbl is natural I think, however, stmt.tbls[i] is strange here.